src/main/java/picard/illumina/parser/readers/TileMetricsOutReader.java

```diff
@@ -27,30 +27,54 @@
  * byte 6-9 (float)          = metrics value, see Theory of RTA document by Illumina for definition
  */
 public class TileMetricsOutReader implements Iterator<TileMetricsOutReader.IlluminaTileMetrics> {
-    private static final int HEADER_SIZE = 2;
-    private static final int EXPECTED_RECORD_SIZE = 10;
-    private static final int EXPECTED_VERSION = 2;
-
+    private static final int VERSION_TWO_RECORD_SIZE = 10;
+    private static final int VERSION_TWO_HEADER_SIZE = 2;
+    private static final int VERSION_THREE_RECORD_SIZE = 15;
+    private static final int VERSION_THREE_HEADER_SIZE = 6;
     private final BinaryFileIterator<ByteBuffer> bbIterator;
+    private float density;
+    private int version;
 
     /**
      * Return a TileMetricsOutReader for the specified file
      * @param tileMetricsOutFile The file to read
+     * @param version
      */
-    public TileMetricsOutReader(final File tileMetricsOutFile) {
-        bbIterator = MMapBackedIteratorFactory.getByteBufferIterator(HEADER_SIZE, EXPECTED_RECORD_SIZE, tileMetricsOutFile);
+    public TileMetricsOutReader(final File tileMetricsOutFile, int version) {
+
+        int headerSize;
+        int recordSize;
+
+        switch (version) {
+            case 2:
+                headerSize = VERSION_TWO_HEADER_SIZE;
+                recordSize = VERSION_TWO_RECORD_SIZE;
+                break;
+            case 3:
+                headerSize = VERSION_THREE_HEADER_SIZE;
+                recordSize = VERSION_THREE_RECORD_SIZE;
+                break;
+            default:
+                throw new PicardException("Unknown version " + version);
```

Comment:
Version should be an enum to avoid this issue. you could store the sizes there as well.